src/lib/input/_input-theme.scss

```diff
@@ -52,26 +52,28 @@
 
   .mat-input-underline {
     border-color: $input-underline-color;
+  }
 
-    .mat-input-ripple {
-      background-color: $input-underline-focused-color;
+  .mat-input-ripple {
+    background-color: $input-underline-focused-color;
 
-      &.mat-accent {
-        background-color: $input-underline-color-accent;
-      }
-      &.mat-warn {
-        background-color: $input-underline-color-warn;
-      }
+    &.mat-accent {
+      background-color: $input-underline-color-accent;
+    }
+
+    &.mat-warn {
+      background-color: $input-underline-color-warn;
     }
   }
 
   // Styling for the error state of the input container. Note that while the same can be
   // achieved with the ng-* classes, we use this approach in order to ensure that the same
   // logic is used to style the error state and to show the error messages.
   .mat-input-invalid {
-    .mat-input-placeholder,
-    .mat-placeholder-required {
-      color: $input-underline-color-warn;
+    .mat-input-placeholder {
+      &, &.mat-accent, &.mat-float .mat-placeholder-required {
```

Comment:
can you put each selector on its own line (makes it obvious that there's multiple and not one complex one).

also I think we can't have rules that end with `&` in our theme files, so maybe:

```scss
.mat-input-placeholder {
  color: $input-underline-color-warn;

  &.mat-accent,
  &.mat-float .mat-placeholder-required {
    color: $input-underline-color-warn;
  }
}
```